app/model/entity/SolutionZipFile.php

```diff
@@ -35,4 +35,31 @@ public function getFile(FileStorageManager $manager): ?IImmutableFile
     {
         return $manager->getSolutionFile($this->getSolution()); // this will return entire archive
     }
+
+    public function getNestedFile(FileStorageManager $manager, string $entryName): ?IImmutableFile
+    {
+        return $manager->getSolutionFile($this->getSolution(), $entryName);
+    }
+
+    private $zipEntries = null;
+
+    public function prepareExtendedSerializationData(FileStorageManager $manager): void
```

Comment:
I believe we introduced concept of views, which constructs JSON responses, in this case it seems like it would be much better solution, than filling entity with response data...